pkg/executor/docker/executor.go

```diff
@@ -43,15 +40,18 @@ const (
 
 type Executor struct {
 	// used to allow multiple docker executors to run against the same docker server
-	ID          string
+	ID string
+
+	handlers generic.SyncMap[string, *executionHandler]
```

Comment:
I'm guessing it's execution id -> handler, but is it worth a brief comment?  
In general though I like this pattern, it makes it much clearer that `one executor -> many active executions` and it wasn't super clear from our code before.

Reply:
I have added a comment describing the maps purpose. You are correct, executionID -> handler.